tests/testthat/test-utils.R

```diff
@@ -0,0 +1,120 @@
+box::use(
+  shiny[testServer],
+  testthat[...],
+  waiter[waiter_preloader],
+  shiny[img, div, br, tagList],
+  shiny.semantic[dropdown_input],
+  glue[glue],
+  plotly[add_trace, add_bars, layout, config, plotlyProxy, plotlyProxyInvoke],
+  magrittr[`%>%`],
+  dplyr[n, select, mutate],
+  purrr[map, pmap]
+)
+
+box::use(
+  app/logic/utils[
+    loading_screen,
+    make_dropdown,
+    update_timeline_colors,
+    customize_axes,
+    add_medal_trace,
+    add_podium_bar,
+    make_flag_data,
+    make_flags,
+    update_podium_flags,
+    reset_podium_flags
+  ]
+)
+
+# loading_screen --------------------------------------------------------------
+test_that("loading_screen: invalid param 'text' is handled", {
+  expect_error(loading_screen(text = "", bkg_color = "white"))
```

Comment:
I think a couple of `expect_error()` calls can be a nice piece of testing, but I think it doesn't add much value to do it this comprehensively. In particular, `loading_screen()` is used once in the whole application and it is unlikely to change (this function defines a fixed part of the page and uses a hardcoded logo).

It's not an easy task to write valuable tests. It might help to ask yourself questions like: Which parts of the application are sketchy / likely to contain mistakes? Which parts don't I understand?

Reply:
Good point, you're right. Did it because I'm accustomed to testing everything for test coverage + I like making sure one cannot abuse types/parameters. Next time I will approach this as you suggested